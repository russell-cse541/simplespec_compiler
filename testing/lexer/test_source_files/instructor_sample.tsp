// This is a comment

namespace  Main;

using Locale;

scalar ID = ASCIIString {
    requires: value.length();
}

model User {
    id: ID;
    name: String;
    region: Locale::Country;
    team: List<User>;
}

function checkit(u: User): Bool {
    let a:String = a<Yes<Bool>>();
    return true;
}

