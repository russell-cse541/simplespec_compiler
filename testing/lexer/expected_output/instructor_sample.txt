
namespace              TagNum: 258    Line: 3     Col: 1          
Main                   TagNum: 303    Line: 3     Col: 12         
;                      TagNum: 302    Line: 3     Col: 16         
using                  TagNum: 259    Line: 5     Col: 1          
Locale                 TagNum: 303    Line: 5     Col: 7          
;                      TagNum: 302    Line: 5     Col: 13         
scalar                 TagNum: 265    Line: 7     Col: 1          
ID                     TagNum: 303    Line: 7     Col: 8          
=                      TagNum: 291    Line: 7     Col: 11         
ASCIIString            TagNum: 303    Line: 7     Col: 13         
{                      TagNum: 281    Line: 7     Col: 25         
requires               TagNum: 264    Line: 8     Col: 5          
:                      TagNum: 299    Line: 8     Col: 13         
value                  TagNum: 305    Line: 8     Col: 15         
.                      TagNum: 298    Line: 8     Col: 20         
length                 TagNum: 305    Line: 8     Col: 21         
(                      TagNum: 277    Line: 8     Col: 27         
)                      TagNum: 278    Line: 8     Col: 28         
>                      TagNum: 283    Line: 8     Col: 30         
0n                     TagNum: 306    Line: 8     Col: 32         
;                      TagNum: 302    Line: 8     Col: 34         
}                      TagNum: 282    Line: 9     Col: 1          
model                  TagNum: 261    Line: 11    Col: 1          
User                   TagNum: 303    Line: 11    Col: 7          
{                      TagNum: 281    Line: 11    Col: 12         
id                     TagNum: 305    Line: 12    Col: 5          
:                      TagNum: 299    Line: 12    Col: 7          
ID                     TagNum: 303    Line: 12    Col: 9          
;                      TagNum: 302    Line: 12    Col: 11         
name                   TagNum: 305    Line: 13    Col: 5          
:                      TagNum: 299    Line: 13    Col: 9          
String                 TagNum: 303    Line: 13    Col: 11         
;                      TagNum: 302    Line: 13    Col: 17         
region                 TagNum: 305    Line: 14    Col: 5          
:                      TagNum: 299    Line: 14    Col: 11         
Locale                 TagNum: 303    Line: 14    Col: 13         
::                     TagNum: 300    Line: 14    Col: 19         
Country                TagNum: 303    Line: 14    Col: 21         
;                      TagNum: 302    Line: 14    Col: 28         
team                   TagNum: 305    Line: 15    Col: 5          
:                      TagNum: 299    Line: 15    Col: 9          
List                   TagNum: 303    Line: 15    Col: 11         
<                      TagNum: 284    Line: 15    Col: 15         
User                   TagNum: 303    Line: 15    Col: 16         
>                      TagNum: 283    Line: 15    Col: 20         
;                      TagNum: 302    Line: 15    Col: 21         
}                      TagNum: 282    Line: 16    Col: 1          
function               TagNum: 266    Line: 18    Col: 1          
checkit                TagNum: 305    Line: 18    Col: 10         
(                      TagNum: 277    Line: 18    Col: 17         
u                      TagNum: 305    Line: 18    Col: 18         
:                      TagNum: 299    Line: 18    Col: 19         
User                   TagNum: 303    Line: 18    Col: 21         
)                      TagNum: 278    Line: 18    Col: 25         
:                      TagNum: 299    Line: 18    Col: 26         
Bool                   TagNum: 303    Line: 18    Col: 28         
{                      TagNum: 281    Line: 18    Col: 33         
return                 TagNum: 267    Line: 19    Col: 5          
u                      TagNum: 305    Line: 19    Col: 12         
.                      TagNum: 298    Line: 19    Col: 13         
friends                TagNum: 305    Line: 19    Col: 14         
.                      TagNum: 298    Line: 19    Col: 21         
allOf                  TagNum: 305    Line: 19    Col: 22         
(                      TagNum: 277    Line: 19    Col: 27         
pred                   TagNum: 269    Line: 19    Col: 28         
<                      TagNum: 284    Line: 19    Col: 32         
User                   TagNum: 303    Line: 19    Col: 33         
>                      TagNum: 283    Line: 19    Col: 37         
(                      TagNum: 277    Line: 19    Col: 38         
f                      TagNum: 305    Line: 19    Col: 39         
:                      TagNum: 299    Line: 19    Col: 40         
User                   TagNum: 303    Line: 19    Col: 42         
)                      TagNum: 278    Line: 19    Col: 46         
:                      TagNum: 299    Line: 19    Col: 47         
Bool                   TagNum: 303    Line: 19    Col: 49         
{                      TagNum: 281    Line: 19    Col: 54         
u                      TagNum: 305    Line: 19    Col: 55         
.                      TagNum: 298    Line: 19    Col: 56         
region                 TagNum: 305    Line: 19    Col: 57         
==                     TagNum: 293    Line: 19    Col: 64         
f                      TagNum: 305    Line: 19    Col: 67         
.                      TagNum: 298    Line: 19    Col: 68         
region                 TagNum: 305    Line: 19    Col: 69         
}                      TagNum: 282    Line: 19    Col: 75         
)                      TagNum: 278    Line: 19    Col: 76         
;                      TagNum: 302    Line: 19    Col: 77         
}                      TagNum: 282    Line: 20    Col: 1          
There were no unrecognized tokens
Token Count : 87
